updater/fetchers/rhel/rhel.go

```diff
@@ -34,326 +29,98 @@ const (
 	// Before this RHSA, it deals only with RHEL <= 4.
 	firstRHEL5RHSA      = 20070044
 	firstConsideredRHEL = 5
-
-	ovalURI        = "https://www.redhat.com/security/data/oval/"
-	rhsaFilePrefix = "com.redhat.rhsa-"
-	updaterFlag    = "rhelUpdater"
 )
 
 var (
-	ignoredCriterions = []string{
-		" is signed with Red Hat ",
-		" Client is installed",
-		" Workstation is installed",
-		" ComputeNode is installed",
-	}
-
 	rhsaRegexp = regexp.MustCompile(`com.redhat.rhsa-(\d+).xml`)
-
-	log = capnslog.NewPackageLogger("github.com/coreos/clair", "updater/fetchers/rhel")
+	log        = capnslog.NewPackageLogger("github.com/coreos/clair", "updater/fetchers/rhel")
 )
 
-type oval struct {
-	Definitions []definition `xml:"definitions>definition"`
+func init() {
+	rhelInfo := &RHELInfo{}
+	updater.RegisterFetcher(rhelInfo.DistName(),
+		&oval.OvalFetcher{OsInfo: rhelInfo})
 }
 
-type definition struct {
-	Title       string      `xml:"metadata>title"`
-	Description string      `xml:"metadata>description"`
-	References  []reference `xml:"metadata>reference"`
-	Criteria    criteria    `xml:"criteria"`
+// RHELInfo implements oval.OsInfo interface
+// See oval.OsInfo for more info on what each method is
+type RHELInfo struct {
 }
 
-type reference struct {
-	Source string `xml:"source,attr"`
-	URI    string `xml:"ref_url,attr"`
+func (f *RHELInfo) DistFile(item string) string {
+	rhsaFilePrefix := "com.redhat.rhsa-"
+	return f.OvalURI() + rhsaFilePrefix + item + ".xml"
 }
 
-type criteria struct {
-	Operator   string      `xml:"operator,attr"`
-	Criterias  []*criteria `xml:"criteria"`
-	Criterions []criterion `xml:"criterion"`
+func (f *RHELInfo) SecToken() string {
+	return "RHSA"
 }
 
-type criterion struct {
-	Comment string `xml:"comment,attr"`
+func (f *RHELInfo) IgnoredCriterions() []string {
+	return []string{
+		" is signed with Red Hat ",
+		" Client is installed",
+		" Workstation is installed",
+		" ComputeNode is installed",
+	}
 }
 
-// RHELFetcher implements updater.Fetcher and gets vulnerability updates from
-// the Red Hat OVAL definitions.
-type RHELFetcher struct{}
-
-func init() {
-	updater.RegisterFetcher("Red Hat", &RHELFetcher{})
+func (f *RHELInfo) OvalURI() string {
+	return "https://www.redhat.com/security/data/oval/"
 }
 
-// FetchUpdate gets vulnerability updates from the Red Hat OVAL definitions.
-func (f *RHELFetcher) FetchUpdate(datastore database.Datastore) (resp updater.FetcherResponse, err error) {
-	log.Info("fetching Red Hat vulnerabilities")
-
-	// Get the first RHSA we have to manage.
-	flagValue, err := datastore.GetKeyValue(updaterFlag)
-	if err != nil {
-		return resp, err
-	}
-	firstRHSA, err := strconv.Atoi(flagValue)
```

Comment:
This ensures that Clair doesn't try to re-parse RHSAs that have already been processed and stored - in order to make the updates faster.